spring-boot/src/main/java/org/springframework/boot/jta/narayana/NarayanaBeanFactoryPostProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.jta.narayana;
+
+import javax.transaction.TransactionManager;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.core.Ordered;
+
+/**
+ * {@link BeanFactoryPostProcessor} to automatically setup correct beans
+ * ordering.
+ *
+ * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
+ */
+public class NarayanaBeanFactoryPostProcessor
+		implements BeanFactoryPostProcessor, Ordered {
+
+	private static final String[] NO_BEANS = {};
+
+	private static final int ORDER = Ordered.LOWEST_PRECEDENCE;
+
+	@Override
+	public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
+			throws BeansException {
+
+		String[] transactionManagers = beanFactory.getBeanNamesForType(
+				TransactionManager.class, true, false);
+
+		for (String transactionManager : transactionManagers) {
+			addTransactionManagerDependencies(beanFactory, transactionManager);
+		}
+	}
+
+	private void addTransactionManagerDependencies(
+			ConfigurableListableBeanFactory beanFactory,
+			String transactionManager) {
+
+		for (String dependentBeanName : getBeanNamesForType(
+				beanFactory, "javax.sql.DataSource")) {
+
+			beanFactory.registerDependentBean(transactionManager, dependentBeanName);
+		}
+
+		for (String dependentBeanName : getBeanNamesForType(
+				beanFactory, "javax.jms.ConnectionFactory")) {
+
+			beanFactory.registerDependentBean(transactionManager, dependentBeanName);
+		}
+	}
+
+	private String[] getBeanNamesForType(
+			ConfigurableListableBeanFactory beanFactory, String type) {
+
+		try {
+			return beanFactory.getBeanNamesForType(Class.forName(type), true, false);
+		}
+		catch (ClassNotFoundException ex) {
+			// Ignore
+		}
+		catch (NoClassDefFoundError ex) {
+			// Ignore
```

Comment:
I probably asked this before but can you modify the comment why its safe to ignore